Tools/YamlCreate.ps1

```diff
@@ -2576,36 +2628,141 @@ if (($script:Option -eq 'QuickUpdateVersion') -and ($ScriptSettings.SuppressQuic
     'Y' { Write-Host -ForegroundColor DarkYellow -Object "`n`nContinuing with Quick Update" }
     'N' { $script:Option = 'New'; Write-Host -ForegroundColor DarkYellow -Object "`n`nSwitched to Full Update Experience" }
     default {
-      Write-Host
-      [Threading.Thread]::CurrentThread.CurrentUICulture = $callingUICulture
-      [Threading.Thread]::CurrentThread.CurrentCulture = $callingCulture
-      exit
+      Invoke-CleanExit
     }
   }
 }
 Write-Host
 
-# Request Package Identifier and Validate
-do {
-  if ((Test-String $PackageIdentifier -IsNull) -or ($script:_returnValue.StatusCode -ne [ReturnValue]::Success().StatusCode)) {
-    Write-Host -ForegroundColor 'Red' $script:_returnValue.ErrorString()
-    Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the Package Identifier, in the following format <Publisher shortname.Application shortname>. For example: Microsoft.Excel'
-    $script:PackageIdentifier = Read-Host -Prompt 'PackageIdentifier' | TrimString
+# Confirm the user undertands the implications of moving package
+if (($script:Option -eq 'MovePackageIdentifier')) {
+  $_menu = @{
+    entries       = @('[Y] Continue moving package'; '*[Q] Exit Script')
+    Prompt        = 'Packages should only be moved between identifiers when necessary. Are you sure you want to continue?'
+    HelpText      = 'This mode should be used with caution. If you are not 100% certain what you are doing, please open an issue at GitHub'
+    HelpTextColor = 'Red'
+    DefaultString = 'Q'
   }
+  switch ( Invoke-KeypressMenu -Prompt $_menu['Prompt'] -Entries $_menu['Entries'] -DefaultString $_menu['DefaultString'] -HelpText $_menu['HelpText'] -HelpTextColor $_menu['HelpTextColor']) {
+    'Y' {
+      # To move a package doesn't require a package version like the other functions of YamlCreate. Therefore, to avoid requesting the information twice
+      # the entirety of the Move packages script happens here. This will then exit the script directly. Additionally, this mode will not create the PRs
+      # for the user, but it will create and push the branches
+      Write-Host; Write-Host
+
+      # Update the ref for upstream master and switch to it to ensure the latest manifest information
+      git fetch upstream master --quiet
```

Comment:
This will fail if user has not set upstream to point to https://github.com/microsoft/winget-pkgs. There is code further down in this script to set that up, can we move it above so that upstream is set before this code's execution?


https://github.com/microsoft/winget-pkgs/blob/e88e70ecbd6bc4fe158d2d0ca94db5a18243d922/Tools/YamlCreate.ps1#L3089-L3096

Reply:
Moved that code up to the top of the script and put the logic for reverting any changes to the user's existing config into the clean exit